### DISCUSSION TITLE
Any difference for milo client to work with Kepware or S7-1500

### DISCUSSION THREAD

**Discussion by weimengxz** [eclipse-milo/milo#750] [General]
As we all know, both Kepware and siemens S7-1500 can work as OPC-UA server. So if any difference for milo client to communicate with them both?

**Comment by kevinherron** [comment]
The address space and NodeIds will be totally different but fundamentally there’s no reason a Milo-based OPC UA client can’t communicate with either.

I know from firsthand experience it works with both of these servers.
